storage/replicate_queue.go

```diff
@@ -162,17 +165,31 @@ func (rq *replicateQueue) process(
 		log.Event(ctx, "removing a replica")
 		// We require the lease in order to process replicas, so
 		// repl.store.StoreID() corresponds to the lease-holder's store ID.
-		removeReplica, err := rq.allocator.RemoveTarget(desc.Replicas, repl.store.StoreID())
+		removeReplica, err := rq.allocator.RemoveTarget(desc.Replicas, 0)
```

Comment:
This is the only non-test user of the second argument to RemoveArgument, so if we do this, we could remove that argument completely. However, as I said in #9462, we need to have a strong preference for rebalancing ranges for which we don't have the lease, or we reintroduce #5737 and the associated availability problems.